mars/services/scheduling/supervisor/queueing.py

```diff
@@ -186,3 +192,24 @@ def remove_queued_subtasks(self, subtask_ids: List[str]):
         for stid in subtask_ids:
             self._stid_to_bands.pop(stid, None)
             self._stid_to_items.pop(stid, None)
+
+    async def balance_queued_subtasks(self, from_band: Tuple = None):
+        # record length of band queues of one specific band or all bands
+        band_num_queued_subtasks = \
+            {from_band: len(self._band_queues[from_band])} if from_band is not None else \
+            {band: len(queue) for band, queue in self._band_queues.items()}
```

Comment:
It is not appropriate to assign subtasks that are already assigned in existing bands.

Reply:
We think that if some band is temporarily marked as STOPPING while there are still subtasks left on the according task queue, they need moving to other ready bands. `{from_band: len(self._band_queues[from_band])}` currently come from line 135. I'm wondering where the inappropriateness lies.